ansible/roles/containers/tasks/atfncat.yaml

```diff
@@ -0,0 +1,56 @@
+- name: Create atfncat container
+  vars:
+    base_container_image : "{{ common_container_dir }}/{{ base_container_name }}.sif"
+    container_name: "atfncat"
+    container_recipe: "{{ common_container_recipe_dir }}/{{ container_name }}.def"
+    container_image: "{{ common_container_dir }}/{{ container_name }}.sif"
+    atfncat_module_dir: "{{ common_modules }}/atfncat"
+    atfncat_binary_dir: "{{ common_container_dir }}/atfncat"
+  tags:
+    - "atfncat"
+    - never
+  block:
+  - name: "Check for existence of {{ base_container_image }} container"
+    stat:
+      path: "{{ base_container_image }}"
+    register: base_container_exists
+  - name: "Missing {{ base_container_image }}"
+    fail:
+      msg: "The container {{ base_container_image }} is missing."
+    when: not base_container_exists.stat.exists
+  - name: Create atfncat container
+    when: base_container_exists.stat.exists
+    block:
+      - name: Copy atfncat recipe  {{ container_recipe }}
+        template:
+          src: "atfncat.def"
+          dest: "{{ container_recipe }}"
+      - name: Build singularity container
+        command: "{{ singularity }} build {{ container_image }} {{ common_container_recipe_dir }}/{{ container_name }}.def"
+        args:
+            creates: "{{ container_image }}"
+        become: true
+        environment:
+            SINGULARITY_TMPDIR: "{{ lookup('env','SINGULARITY_TMPDIR') | default('/dev/shm') }}"
+      - name: Ensure atfncat binary directory exists
+        file:
+          path: "{{ atfncat_binary_dir }}"
+          state: directory
+          mode: u=rwx,g=rwx,o=rx
+      - name: Create atfncat binary
+        template:
+          src: "atfncat.sh"
+          dest: "{{ atfncat_binary_dir }}/psrcat"
+          mode: 0755
+
+  - name: Install {{ container_name }} module
+    block:
+      - name: Check module dir
+        file:
+          path: "{{ atfncat_module_dir }}"
+          state: directory
+          mode: u=rwx,g=rwx,o=rx
+      - name: Install atfncat module file
+        template:
+          src: atfncat.lua
+          dest: "{{ atfncat_module_dir }}/default.lua"
```

Comment:
I might be wrong, but I think in most cases users probably don't care whether it's in a container or not?

Reply:
It would also be hard to indicate this when there are multiple versions of the software.
Hmm... maybe it would be best to just specify the software version here for psrcat. 
I remember you said lmod has cool rules about how to pick the default version anyway. So they could then just do `module load atfncat`
 
[lmod default version rules](https://lmod.readthedocs.io/en/latest/060_locating.html#:~:text=The%20user%20may%20specify%20N%20%2Fdefault%20as%20N%2FV,then%20continues%20as%20if%20it%20was%20never%20there.
)

Reply:
Interesting link :-). I think I've always just glossed over it and marvelled at how well it works :-p.

Yeah, mixed feelings about how to do this. So for R, there is both compiled and container versions so kind feels like _maybe_ it's worth indicating. You're probably right though, a version number usually sufficient.